pkg/assembler/backends/ent/backend/dependency.go

```diff
@@ -116,6 +116,28 @@ func getIsDepObject(q *ent.DependencyQuery) *ent.DependencyQuery {
 		Order(ent.Asc(dependency.FieldID))
 }
 
+// getIsDepObjectWithoutEdges is used recreate the isDependency object without eager loading the edges
+func getIsDepObjectWithoutEdges(q *ent.DependencyQuery) *ent.DependencyQuery {
+	return q.
+		Order(ent.Asc(dependency.FieldID))
+}
+
+// deleteIsDependency is called by hasSBOM to delete the isDependency nodes that are part of the hasSBOM
+func (b *EntBackend) deleteIsDependency(ctx context.Context, isDependencyID uuid.UUID) error {
+	_, txErr := WithinTX(ctx, b.client, func(ctx context.Context) (*string, error) {
```

Comment:
to delete the SBOM this will have to be done in multiple transactions

Reply:
Well there might be a better way. Blocking comment for now.

Reply:
updated. So now it will be limited to 3 transactions (delete all isDeps, delete all isOcurr, and delete hasSBOM)